### ISSUE TITLE
Issue with Custom Image Delegate - Image from memory

### ISSUE THREAD

**Issue by VenkataPakalapati** [memspace/zefyr#330]
Custom Image Delegate keeps calling **buildImage** method.

here is my custom delegate code

```
class CustomImageDelegate extends ZefyrImageDelegate<ImageSource> {
  @override
  Widget buildImage(BuildContext context, String imageSource) {
    **print("building image");**
    var decodedbase64 = base64Decode(imageSource);
    return Image.memory(decodedbase64);
  }

  @override
  Future<String> pickImage(ImageSource source) async {
    final file = await ImagePicker.pickImage(source: source);
    if (file == null) return null;
    return base64Encode(file.readAsBytesSync());
  }

  @override
  ImageSource get cameraSource => ImageSource.camera;

  @override
  ImageSource get gallerySource => ImageSource.gallery;
}
```

link to video -- https://drive.google.com/file/d/1dqbrl1nt9Ra6qpDtrAKioaNvqU9CD0O_/view?usp=sharing

console: 

![image](https://user-images.githubusercontent.com/17863587/83070376-610c5680-a039-11ea-8893-e7daf756e6e2.png)

**Comment by cgestes** [comment]
everytime the cursor blink?

If that's the case, I guess you need to find a way to avoid the call to 'buildImage' to be expensive.

**Comment by VenkataPakalapati** [comment]
yes but it only happens if i try to return an image from memory. basically this is what i was trying to do

- in **pickImage** convert picked image to base64 encoded string
in **buildImage** decode that string and return a MemoryImage

If i switch to using **FileImage**, there is no issue

**Comment by krishnakumarcn** [comment]
Hey @VenkataPakalapati ,
Any solution to this?

**Comment by doducvuong14061990** [comment]
Have you fixed the error yet?
please help me.

**Comment by doducvuong14061990** [comment]
> Custom Image Delegate keeps calling **buildImage** method.
> 
> here is my custom delegate code
> 
> ```
> class CustomImageDelegate extends ZefyrImageDelegate<ImageSource> {
>   @override
>   Widget buildImage(BuildContext context, String imageSource) {
>     **print("building image");**
>     var decodedbase64 = base64Decode(imageSource);
>     return Image.memory(decodedbase64);
>   }
> 
>   @override
>   Future<String> pickImage(ImageSource source) async {
>     final file = await ImagePicker.pickImage(source: source);
>     if (file == null) return null;
>     return base64Encode(file.readAsBytesSync());
>   }
> 
>   @override
>   ImageSource get cameraSource => ImageSource.camera;
> 
>   @override
>   ImageSource get gallerySource => ImageSource.gallery;
> }
> ```
> 
> ![image](https://user-images.githubusercontent.com/17863587/83077609-c5351780-a045-11ea-91cd-4e3388f21342.png)
> 
> link to video -- https://drive.google.com/file/d/1dqbrl1nt9Ra6qpDtrAKioaNvqU9CD0O_/view?usp=sharing
> 
> console:
> 
> ![image](https://user-images.githubusercontent.com/17863587/83070376-610c5680-a039-11ea-8893-e7daf756e6e2.png)

Have you fixed the error yet?
please help me.
